tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java

```diff
@@ -59,21 +59,16 @@ public void setRollbackRules(Set<RollbackRule> rollbackRules) {
     }
 
     public boolean rollbackOn(Throwable ex) {
-
         RollbackRule winner = null;
-        int deepest = Integer.MAX_VALUE;
-
         if (CollectionUtils.isNotEmpty(rollbackRules)) {
             winner = NoRollbackRule.DEFAULT_NO_ROLLBACK_RULE;
             for (RollbackRule rule : this.rollbackRules) {
                 int depth = rule.getDepth(ex);
-                if (depth >= 0 && depth < deepest) {
-                    deepest = depth;
+                if (depth >= 0) {
                     winner = rule;
```

Comment:
rollbackRules is LinkedHashSet .  in this ,if i add break ,RollbackRule will be a Higher priority. 
but now, it just like as it was : Loop will execute to end ,and if RollbackRule and NoRollBackRule has same Exception, noRollBackRule will will take effect.

Reply:
same Exception refers to the relationship of inheritance

Reply:
`if RollbackRule and NoRollBackRule has same Exception` Regardless of this scenario, users need to be responsible for their own actions. If you say both yes and no about a thing, it is impossible for the code to determine whether it is or not.